pkg/controller/provider/container/ovirt/client.go

```diff
@@ -219,3 +221,138 @@ func GetInsecureSkipVerifyFlag(secret *core.Secret) bool {
 
 	return insecureSkipVerify
 }
+
+func TestDownloadOvfStore(secret *core.Secret) (ok bool, err error) {
+
+	// Create the connection to the server:
+	conn, err := ovirtsdk.NewConnectionBuilder().
+		URL(string(secret.Data["url"])).
+		Username(string(secret.Data["user"])).
+		Password(string(secret.Data["password"])).
+		Insecure(true).
+		Compress(true).
+		Timeout(time.Second * 30).
+		Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	defer conn.Close()
+
+	//Find OVF STORE ID
+	disksService := conn.SystemService().DisksService()
+	diskResponse, err := disksService.List().Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	disks, ok := diskResponse.Disks()
+	var diskIdOVF string
+
+	if ok {
+		//get OVF STORE ID
+		for _, disk := range disks.Slice() {
+			if diskName, ok := disk.Name(); ok {
+				if diskName == "OVF_STORE" {
+					if diskId, ok := disk.Id(); ok {
+						diskIdOVF = diskId
+					}
+					break
+				}
+			}
+		}
+	}
+
+	//in case the OVF STORE is locked wait fo OK status befor stating the image transfer
+	conn.WaitForDisk(diskIdOVF, ovirtsdk.DISKSTATUS_OK, 20*time.Second)
+
+	// Prepare image transfer request
+	transfersService := conn.SystemService().ImageTransfersService()
+	transfer := transfersService.Add()
+	image, err := ovirtsdk.NewImageBuilder().Id(diskIdOVF).Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	imageTransfer, err := ovirtsdk.NewImageTransferBuilder().Image(
+		image,
+	).Direction(
+		ovirtsdk.IMAGETRANSFERDIRECTION_DOWNLOAD,
+	).Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Initialize image transfer and lock the disk
+	transfer.ImageTransfer(imageTransfer)
+	transferReq, err := transfer.Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	currImageTransfer, ok := transferReq.ImageTransfer()
+
+	imageTransferId, ok := currImageTransfer.Id()
+
+	for {
+		if currImageTransfer.MustPhase() == ovirtsdk.IMAGETRANSFERPHASE_INITIALIZING {
+			time.Sleep(1 * time.Second)
+			currImageTransferReq, errIT := conn.SystemService().ImageTransfersService().ImageTransferService(imageTransferId).Get().Send()
+			if err != nil {
+				err = liberr.Wrap(errIT)
+				return
+			}
+			currImageTransfer, ok = currImageTransferReq.ImageTransfer()
+		} else {
+			break
+		}
+	}
+
+	caCert := secret.Data["cacert"]
+	caCertPool := x509.NewCertPool()
+	ok = caCertPool.AppendCertsFromPEM(caCert)
+	if !ok {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Create http client
+	tlsConfig := &tls.Config{
+		RootCAs: caCertPool,
+	}
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	client := &http.Client{Transport: transport}
+
+	// Parse the url for the CA test http request using the ProxyUrl from the image transfer
```

Comment:
I used the same endpoint as configured in the ovirt engine for the same test, IIRC I originally tried to use the transfer URL but it didn't return the desired response.

Reply:
I see
@bennyz do we still configure the proxy in newer oVirt versions?..

Reply:
After testing with the lab env yesterday seems that this check is really not the best (returning OK in cases it should be false)  so it does not cover all cases. 
I switched the proxy URL to the transfer URL and now it will directly test on the host, in case the certificate is missing the request will be failed to send and the error message will give us an indication about that.